### PR TITLE
feat: Implement Authentication on Admin Routes

### PR DESCRIPTION
**Pull request by jinyoungbang**
## What does this PR do?

- Implements Authentication using `next-auth` with a Google Provider
- Implement user dropdown
- Refactored layout (abstracted the flow) to ensure that all admin routes are protected from client side
- Connected vault validation by using `signIn` callback
- Create custom auth error page

## Type of change

- [ ] Fix: Bug fix (non-breaking change which fixes an issue)
- [x] Refactor: Any code refactoring
- [ ] Chore: technical debt, workflow improvements
- [x] Feature: New feature (non-breaking change which adds functionality)
- [ ] Documentation: This change requires a documentation update

## Tests Performed

`npm run dev`

## Screenshots

Screen Shot 2023-09-22 at 10 03 23 PM

Screen Shot 2023-09-22 at 10 03 51 PM

Screenshot 2023-09-24 at 7 00 11 PM

### CODE REVIEW

**Review by yoontsy2000 — COMMENTED** [review]
Line `14` under `app/admin/layout.tsx`.

Is the empty return function necessary? Just curious.

File: app/authError/page.tsx
Comment (yoontsy2000) [inline app/authError/page.tsx:1]:
```
  if (errorType === "AccessDenied") {
    errorMessage = "Access Denied";
  } else {
    errorMessage = errorType;
  }
```

Not sure if the conditional statement is needed.

Comment (jinyoungbang) [reply app/authError/page.tsx:1]: We'll implement different views or different errors -> let's keep it as a template for now

File: components/admin/NavSidebar.tsx
Comment (yoontsy2000) [inline components/admin/NavSidebar.tsx:1]: Remove `Line 69`

File: package.json
Comment (yoontsy2000) [inline package.json:1]: Do we no longer need the plugin?

**Review by yoontsy2000 — CHANGES_REQUESTED** [review]
Made comments on a few files.

File: package.json
Comment (jinyoungbang) [reply package.json:1]: Yup, I think we should be fine - that's what caused the errors to happen on local because of arm/x86

**Review by yoontsy2000 — APPROVED** [review]
Looks good! You fixed the issues I've raised.

**Pull request merged**